aws/data_source_aws_instances_test.go

```diff
@@ -58,108 +62,73 @@ func TestAccAWSInstancesDataSource_instance_state_names(t *testing.T) {
 	})
 }
 
-const testAccInstancesDataSourceConfig_ids = `
-data "aws_ami" "ubuntu" {
-  most_recent = true
-
-  filter {
-    name   = "name"
-    values = ["ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-*"]
-  }
-
-  filter {
-    name   = "virtualization-type"
-    values = ["hvm"]
-  }
-
-  owners = ["099720109477"] # Canonical
-}
-
+func testAccInstancesDataSourceConfig_ids(rName string) string {
+	return composeConfig(
+		testAccLatestAmazonLinuxHvmEbsAmiConfig(),
+		testAccAvailableEc2InstanceTypeForRegion("t3.micro", "t2.micro"),
+		fmt.Sprintf(`
 resource "aws_instance" "test" {
   count         = 3
-  ami           = data.aws_ami.ubuntu.id
-  instance_type = "t2.micro"
+  ami           = data.aws_ami.amzn-ami-minimal-hvm-ebs.id
+  instance_type = data.aws_ec2_instance_type_offering.available.instance_type
 
   tags = {
-    Name = "TfAccTest"
+    Name = %q
   }
 }
 
 data "aws_instances" "test" {
   filter {
     name = "instance-id"
     values = [
-      aws_instance.test.*.id[0],
-      aws_instance.test.*.id[1],
-      aws_instance.test.*.id[2],
+      aws_instance.test[0].id,
+      aws_instance.test[1].id,
+      aws_instance.test[2].id,
     ]
   }
 }
-`
-
-func testAccInstancesDataSourceConfig_tags(rInt int) string {
-	return fmt.Sprintf(`
-data "aws_ami" "ubuntu" {
-  most_recent = true
-
-  filter {
-    name   = "name"
-    values = ["ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-*"]
-  }
-
-  filter {
-    name   = "virtualization-type"
-    values = ["hvm"]
-  }
-
-  owners = ["099720109477"] # Canonical
+`, rName))
 }
 
+func testAccInstancesDataSourceConfig_tags(rName string) string {
+	return composeConfig(
+		testAccLatestAmazonLinuxHvmEbsAmiConfig(),
+		testAccAvailableEc2InstanceTypeForRegion("t3.micro", "t2.micro"),
+		fmt.Sprintf(`
 resource "aws_instance" "test" {
   count         = 2
-  ami           = data.aws_ami.ubuntu.id
-  instance_type = "t2.micro"
+  ami           = data.aws_ami.amzn-ami-minimal-hvm-ebs.id
+  instance_type = data.aws_ec2_instance_type_offering.available.instance_type
 
   tags = {
-    Name      = "tf-acc-test-ec2-instance-data-source-%d"
-    SecondTag = "tf-acc-test-ec2-instance-data-source-%d"
+    Name      = %[1]q
+    SecondTag = "%[1]s-2"
   }
 }
 
 data "aws_instances" "test" {
   instance_tags = {
     Name      = aws_instance.test[0].tags["Name"]
-    SecondTag = aws_instance.test[1].tags["Name"]
+    SecondTag = aws_instance.test[1].tags["SecondTag"]
```

Comment:
just a q, no action really needed: if the `tags` block above is the same for the resource at index 0 and index 1 (aws_instance.test[0], aws_instance.test[1]), this could probably reference the resource at index 0 (aws_instance.test[0].tags["SecondTag"] )?

Reply:
You're right!